source/opcode.cpp

```diff
@@ -173,10 +183,12 @@ void spvInstructionCopy(const uint32_t* words, const SpvOp opcode,
 const char* spvOpcodeString(const SpvOp opcode) {
   // Use the latest SPIR-V version, which should be backward-compatible with all
   // previous ones.
-  for (uint32_t i = 0; i < ARRAY_SIZE(opcodeTableEntries_1_2); ++i) {
-    if (opcodeTableEntries_1_2[i].opcode == opcode)
-      return opcodeTableEntries_1_2[i].name;
+  const auto entries = getOpcodeTableEntries_1_2();
+
+  for (uint32_t i = 0; i < entries.len; ++i) {
```

Comment:
Perhaps create a static map on the first call.

Or if the array is in order then std::binary_search (with a lambda returning opcode) can be used to locate the right index in log(n) time.

Reply:
Reasonable. This function is used fairly frequently; using map should save some time. Will do in another pull request. Thanks for the suggestion!

Reply:
That's https://github.com/KhronosGroup/SPIRV-Tools/issues/502

Reply:
Also, I get complaints about memory leaks.  So the create-on-first call means we keep it for subsequent calls.  And that's viewed as a memory leak.  I'm ok with at least not making things worse for now.

The plan in #502 suggests building the sorted tables at SPIRV-Tools build time, so that we still only need POD data.

Reply:
So @dneto0  has more plans for this change. I'll defer doing that.